ipq807x: use upstreamed CPUFreq NVMEM patches

robimarko opened the pull request:
Most of the CPUFreq NVMEM patches have been upstreamed in an improved way. IPQ8074 support itself is being reviewed upstream currently.

So, let's replace the old v1 patches with upstreamed ones and with the latest one for IPQ8074 support.

Runtime tested on Qnap 301W (IPQ8072A) and Xiaomi AX3600 (IPQ8071A).

Ansuel commented:
@robimarko i know this is a lot to ask but can you by chance put these in generic and maybe also backport for 5.15? 

for 5.15 you just need f02a537357a61e7892587c0f3455f8295cc9075c and 6bdab60d5cbd7996e371305433e943a82cde0d4c and then they correctly apply 

for the cpufreq patch i would still put them in ipq8074 and then move to backports when we manage to make them merged.

robimarko commented:
@Ansuel I guess you want to reuse them for IPQ806x?

Do you want just the SMEM patches or everything except for IPQ8074 support itself?
I can probably backport to 5.15 but I am not testing with 5.15 at all.

robimarko committed:
generic: 6.1: backport Qualcomm CPUFreq NVMEM changes

Currently, IPQ807x is using CPUFreq NVMEM for dealing with different SoC
SKU-s having different frequency limits, and we are keeping the patches
for it in ipq807x target.

However, we managed to upstream a big cleanup of the driver in order to
make it possible for other SMEM based targets to be added to CPUFreq NVMEM.

IPQ806x will be using CPUFreq NVMEM and depends on these changes as well,
so lets put them in generic backport to avoid code duplication.

This replaces the older patches in ipq807x.

Signed-off-by: Robert Marko <robimarko@gmail.com>

robimarko committed:
ipq807x: use updated CPUFreq NVMEM support

Most of the CPUFreq NVMEM patches have been upstreamed in an improved way.
IPQ8074 support itself is being reviewed upstream currently.

Upstreamed patches have been moved to generic backports so that ipq806x can
use them as well, so lets just use the latest version of IPQ8074 support
that is being reviewed upstream.

Runtime tested on Qnap 301W (IPQ8072A) and Xiaomi AX3600 (IPQ8071A).

Signed-off-by: Robert Marko <robimarko@gmail.com>

robimarko commented:
Check it now, it should be what you need for IPQ806x as well.

Honestly, 5.15 is a major PITA to backport stuff for currently so I would prefer to avoid having to deal with it.
